Content.Client/DeltaV/CartridgeLoader/Cartridges/NanoChatUi.cs

```diff
@@ -0,0 +1,43 @@
+using Content.Client.UserInterface.Fragments;
+using Content.Shared.CartridgeLoader;
+using Content.Shared.DeltaV.CartridgeLoader.Cartridges;
+using Robust.Client.UserInterface;
+
+namespace Content.Client.DeltaV.CartridgeLoader.Cartridges;
+
+public sealed partial class NanoChatUi : UIFragment
+{
+    private NanoChatUiFragment? _fragment;
+
+    public override Control GetUIFragmentRoot()
+    {
+        return _fragment!;
+    }
+
+    public override void Setup(BoundUserInterface userInterface, EntityUid? fragmentOwner)
+    {
+        _fragment = new NanoChatUiFragment();
+
+        _fragment.OnMessageSent += (type, number, content, job) =>
+        {
+            SendNanoChatUiMessage(type, number, content, job, userInterface);
+        };
+    }
+
+    public override void UpdateState(BoundUserInterfaceState state)
+    {
+        if (state is NanoChatUiState cast)
+            _fragment?.UpdateState(cast);
+    }
+
+    private static void SendNanoChatUiMessage(NanoChatUiMessageType type,
+        uint? number,
+        string? content,
+        string? job,
+        BoundUserInterface userInterface)
+    {
+        var nanoChatMessage = new NanoChatUiMessageEvent(type, number, content, job);
+        var message = new CartridgeUiMessage(nanoChatMessage);
+        userInterface.SendMessage(message);
+    }
```

Comment:
the BUI is meant to do this not the ui fragment
ui fragment should have a public action and be independent of the BUI

Reply:
cartridges don't have a BUI other than the cartridgeloader one, that's how all other things do it?
https://github.com/DeltaV-Station/Delta-v/blob/3a1c59e6ea0f154da5874185ad6b789a5ce27267/Content.Client/CartridgeLoader/Cartridges/NewsReaderUi.cs
https://github.com/DeltaV-Station/Delta-v/blob/3a1c59e6ea0f154da5874185ad6b789a5ce27267/Content.Client/CartridgeLoader/Cartridges/NotekeeperUi.cs